source/server-selection/server-selection.rst

```diff
@@ -579,55 +579,13 @@ Passing read preference to mongos and load balancers
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 If a server of type Mongos or LoadBalancer is selected for a read operation, the read
-preference is passed to the selected mongos through the use of the
-``SecondaryOk`` wire protocol flag, the ``$readPreference`` query
-modifier or both, according to the following rules.
-
-If the read preference contains **only** a ``mode`` parameter and the mode is
-'primary' or 'secondaryPreferred', for maximum backwards compatibility with
-older versions of mongos, drivers MUST only use the value of the ``SecondaryOk``
-wire protocol flag (i.e. set or unset) to indicate the desired read preference
-and MUST NOT use a ``$readPreference`` query modifier.
-
-Therefore, when sending queries to a mongos or load balancer, the following rules apply:
-
-  - For mode 'primary', drivers MUST NOT set the ``SecondaryOk`` wire protocol flag
-    and MUST NOT use ``$readPreference``
-
-  - For mode 'secondary', drivers MUST set the ``SecondaryOk`` wire protocol flag
-    and MUST also use ``$readPreference``
-
-  - For mode 'primaryPreferred', drivers MUST set the ``SecondaryOk`` wire protocol flag
-    and MUST also use ``$readPreference``
-
-  - For mode 'secondaryPreferred', drivers MUST set the ``SecondaryOk`` wire protocol flag.
-    If the read preference contains a non-empty ``tag_sets`` parameter,
-    ``maxStalenessSeconds`` is a positive integer, or the ``hedge`` parameter is
-    non-empty, drivers MUST use ``$readPreference``; otherwise, drivers MUST NOT
-    use ``$readPreference``
-
-  - For mode 'nearest', drivers MUST set the ``SecondaryOk`` wire protocol flag
-    and MUST also use ``$readPreference``
-
-The ``$readPreference`` query modifier sends the read preference as part of the
-query.  The read preference fields ``tag_sets`` is represented in a ``$readPreference``
-document using the field name ``tags``.
-
-When sending a read operation via OP_QUERY and any ``$`` modifier is used, including the ``$readPreference`` modifier,
-the query MUST be provided using the ``$query`` modifier like so::
-
-    {
-        $query: {
-            field1: 'query_value',
-            field2: 'another_query_value'
-        },
-        $readPreference: {
-            mode: 'secondary',
-            tags: [ { 'dc': 'ny' } ],
-            maxStalenessSeconds: 120,
-            hedge: { enabled: true }
-        }
-    }
+preference is passed to the selected mongos through the use of the ``$readPreference``
+`Global Command Argument`_ according to the following rules:
```

Comment:
Note that [DRIVERS-1918](https://jira.mongodb.org/browse/DRIVERS-1918) was _not_ focused specifically on mongos and load balancers (see: [this comment](https://jira.mongodb.org/browse/DRIVERS-1918?focusedCommentId=4087768&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4087768)). [DRIVERS-1642](https://jira.mongodb.org/browse/DRIVERS-1642) was the original issue pertaining to mongos and it was later marked as a duplicate of DRIVERS-1918.

I'm in favor of updating this section, but I think other references to `SecondaryOk` and `$readPreference` will also need to be updated to ensure _all_ topologies are covered.

Also, since [Passing read preference to mongos and load balancers](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#passing-read-preference-to-mongos-and-load-balancers) is a separate section under [Read Preference](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#read-preference), I think we should also include a reference to it from [Topology type: LoadBalanced](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#topology-type-loadbalanced), as is already done for [Topology Type: Sharded](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#topology-type-sharded).

Reply:
It seems like the section name is incorrect now, and it should just be `Passing read preference to MongoDB servers` or something like that.